src/node/errors.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under the MIT license <LICENSE-MIT
+// https://opensource.org/licenses/MIT> or the Modified BSD license <LICENSE-BSD
+// https://opensource.org/licenses/BSD-3-Clause>, at your option. This file may not be copied,
+// modified, or distributed except according to those terms. Please review the Licences for the
+// specific language governing permissions and limitations relating to use of the SAFE Network
+// Software.
+use std::{
+    fmt::{self, Debug, Formatter},
+    result,
+};
+use thiserror::Error;
+
+/// A specialised `Result` type.
+pub type Result<T, E = Error> = result::Result<T, E>;
+
+/// Error debug struct
+pub struct ErrorDebug<'a, T>(pub &'a Result<T>);
+
+impl<'a, T> Debug for ErrorDebug<'a, T> {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        if let Err(error) = self.0 {
+            write!(f, "{:?}", error)
+        } else {
+            write!(f, "Success")
+        }
+    }
+}
+
+/// Main error type for the crate.
+#[derive(Error, Debug)]
+#[non_exhaustive]
+pub enum Error {
```

Comment:
Why do we even need this here? This error doesn't appear anywhere within `node::Message`.

Reply:
ah that was left from when I had all thee routing structs initially in this PR, I'll remove it now and re-add it when adding all the node structs if needed

Reply:
I don't think ti's going to be needed. This error is for routing **logic**, but this crate should contain only data. But we'll see.